obd/elm327.py

```diff
@@ -290,12 +290,14 @@ def auto_baudrate(self):
             # so write a second one (again so that the lone CR doesn't repeat
             # the previous command)
             self.__port.write(b"\x7F\x7F\r\n")
+            self.__port.write(b"ATZ\r\n")
             self.__port.flush()
+            time.sleep(1)
```

Comment:
Do we really need a sleep? I would prefer not to slowdown anything

Reply:
I'm wondering the same thing.  Currently, with everything else being the same, this sleep is necessary.  If I remove it, the auto baudrate method no longer works with my device; presumably, because future writes/reads are being affected by the previous set of writes/reads and perhaps the time.sleep() gives it the time it needs.

It's kind of hard to explain why this code change works because I still don't understand it 100% myself but I discovered that the STN11xx doesn't exactly respond correctly on the first write attempt.  It requires 2 writes where the 2nd write is a known command.  First command seems to be arbitrary.  Why, exactly, is still TBD.  I'd be glad to show you my debug logs; maybe you have some knowledge about how these chipsets work that I don't.  Nonetheless, I think this is a good first step towards figuring out what's really going on.

Regarding the slowdown.  As far as I know, this method is only invoked once during the initial setup of an obd connection, and only in circumstances where baudrate is not known; so, I don't see cause for huge concern.  Nevertheless, this slowdown can be avoided for normal ELM users if we break this out into two separate auto-baudrate passes; one for the ELM and, subsequently, if not found, one for the STN11xx.  If we were to do that, you and other standard ELM users would be unaffected assuming that it detects your baudrate on the first pass.  Would that work for you in the absence of an even better solution?

Reply:
I do see that the initial connection can sometimes be pretty slow, so I would like to avoid having to slow it down any more.

As it looks like most users are ELM users I think it makes sense to try ELM first and if that doesn't work then fall back. In saying that if we have to repeat pretty much the entire transaction again that doesn't make sense either.

Is there any documentation we can look at to see what is required for the STN chips?

Reply:
Thanks for you questions.  After reading the STN documentation, I think I know why it's not working.  I believe I will have a new commit soon which will be much better.